src/app/myProfile/profileUpdate/ProfileUpdate.tsx

```diff
@@ -0,0 +1,138 @@
+import { GymTheme } from '@/MUITheme/GymTheme';
+import '../MyProfile.scss';
+import { Avatar, Button, TextField, ThemeProvider } from '@mui/material';
+import { deepPurple } from '@mui/material/colors';
+import { useForm } from 'react-hook-form';
+import * as Yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+export interface Profile {
+  firstname: string;
+  lastname: string;
+  username: string;
+  bio: string;
+  location: string;
+}
+
+interface ProfileUpdateProps {
+  cancelAction: () => void;
+  saveAction: (profile: Profile) => void;
+  profile: Profile;
+}
+
+const validationSchema = Yup.object().shape({
+  firstname: Yup.string()
+    .required('Firstname is required')
+    .min(3, 'Firstname must be at least 3 characters')
+    .max(64, 'Firstname must be at most 64 characters'),
+  lastname: Yup.string()
+    .required('Lastname is required')
+    .min(3, 'Lastname must be at least 3 characters')
+    .max(64, 'Lastname must be at most 64 characters'),
+  username: Yup.string()
+    .required('Username is required')
+    .min(3, 'Username must be at least 3 characters')
+    .max(32, 'Username must be at most 32 characters'),
+  location: Yup.string().required('Location is required').max(128, 'Location must be at most 128 characters'),
+  bio: Yup.string().max(340, 'Bio must be at most 340 characters'),
+});
+
+const ProfileUpdate = (props: ProfileUpdateProps) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm({
+    defaultValues: props.profile,
+    resolver: yupResolver(validationSchema),
+  });
+
+  const onSubmit = (data: Profile) => {
+    props.saveAction(data);
+  };
+
+  const onCancel = () => {
+    reset(props.profile);
+    props.cancelAction();
+  };
+
+  return (
+    <ThemeProvider theme={GymTheme}>
+      <form className='container' onSubmit={handleSubmit(onSubmit)}>
+        <Avatar
+          src='https://www.fit.pl/img/2007/02/_max/arnie.jpg'
+          className='avatar-image'
+          sx={{ bgcolor: deepPurple[500] }}
+        >
+          OP
+        </Avatar>
+        <div className='profile-data-container'>
+          <TextField
```

Comment:
maybe it would be nice to create either text-field for whole app with it's style or here create ProfileTextField where you can already include all data that is the same in each:

            fullWidth
            size='small'
            className='form-field'
            variant='filled'

Reply:
There is the same problem - when moving textFields into a separate component, react-form-hook stops inputing the default userProfile data, etc...
So we need to add watching, add onChange eventHandlers, etc...
So we would have the same amount of code in `ProfileUpdate.tsx` + a new component for 50 lines of code